apps/auth/db/db.py

```diff
@@ -24,6 +24,12 @@ def authenticate(self, credentials, ignore_expire=False):
         if not user:
             raise CredentialsAuthError(credentials)
 
+        _user = get_resource_service("users").find_one(req=None, username=credentials.get("username"))
+        if _user.get("user_type") == "external":
+            raise ExternalUserError(
+                message="Oops!This account has been changed to External. External accounts have no login capability."
```

Comment:
I guess you only need 1 place to specify the message, and would be good to use gettext for it